File: mobile/lib/models/exercise.dart
class Exercise {
  final String id;
  final String name;
  final String target;
  final String equipment;
  final String image;

  const Exercise({
    required this.id,
    required this.name,
    required this.target,
    required this.equipment,
    required this.image,
  });

  factory Exercise.fromJson(Map<String, dynamic> json) {
    return Exercise(
      id: json['id'] as String,
      name: json['name'] as String,
      target: json['target'] as String,
      equipment: json['equipment'] as String,
      image: json['image'] as String? ?? '',
    );
  }
}

File: backend/app/models/exercise.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class Exercise(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    name: str
    category: str
    body_part: str
    equipment: str
    instructions: dict[str, str | list[str]]
    instruction_steps: dict[str, list[str]] | list[str]
    muscle_group: str
    secondary_muscles: list[str]
    target: str
    media_id: str
    image: str
    gif_url: str
    attribution: str
    created_at: datetime

class ExerciseSummary(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    name: str
    body_part: str
    target: str
    muscle_group: str
    equipment: str
    image: str
    gif_url: str

class ExerciseListResponse(BaseModel):
    total: int
    page: int
    page_size: int
    pages: int
    muscle: str | None
    search: str | None
    equipment: str | None
    body_part: str | None
    items: list[ExerciseSummary]

class OptionListResponse(BaseModel):
    total: int
    items: list[str]
